cmd/juju/bundle/exportbundle.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package bundle
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/bundle"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+)
+
+// NewexportbundleCommand returns a fully constructed exportbundle command.
+func NewexportbundleCommand() cmd.Command {
+	return modelcmd.Wrap(&exportBundleCommand{}, modelcmd.WrapSkipModelFlags)
+}
+
+type exportBundleCommand struct {
+	modelcmd.ModelCommandBase
+	out cmd.Output
+	api ExportbundleModelAPI
+}
+
+const exportBundleHelpDoc = `
+Exports the current model configuration into a YAML file.
+
+Examples:
+
+    juju export-bundle
+`
+
+// Info implements Command.
+func (c *exportBundleCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "export-bundle",
+		Purpose: "Exports the current model configuration in a YAML file.",
+		Doc:     exportBundleHelpDoc,
+	}
+}
+
+// SetFlags implements Command.
+func (c *exportBundleCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	c.out.AddFlags(f, "yaml", output.DefaultFormatters)
+}
+
+// Init implements Command.
+func (c *exportBundleCommand) Init(args []string) error {
+	modelName := ""
```

Comment:
There should be no positional args. If a user wants a different model they'd say:

```
juju export-bundle -m other
```
And the api connection machinery in the base type takes care of getting the api connection to the correct model.